rust/agama-dbus-server/src/questions.rs

```diff
@@ -165,62 +112,75 @@ impl Questions {
     #[dbus_interface(name = "New")]
     async fn new_question(
         &mut self,
+        class: &str,
         text: &str,
         options: Vec<&str>,
-        default_option: Vec<&str>,
+        default_option: &str,
+        data: HashMap<String, String>,
     ) -> Result<ObjectPath, zbus::fdo::Error> {
         let id = self.last_id;
         self.last_id += 1; // TODO use some thread safety
         let options = options.iter().map(|o| o.to_string()).collect();
-        // TODO: enforce default option and do not use an array
-        let question = GenericQuestion::new(
+        let mut question = questions::GenericQuestion::new(
             id,
+            class.to_string(),
             text.to_string(),
             options,
-            default_option.first().map(|o| o.to_string()),
+            default_option.to_string(),
+            data,
         );
+        self.fill_answer(&mut question);
         let object_path = ObjectPath::try_from(question.object_path()).unwrap();
+        let question_object = GenericQuestionObject(question);
 
         self.connection
             .object_server()
-            .at(question.object_path(), question)
+            .at(object_path.clone(), question_object)
             .await?;
-        self.questions.insert(id, QuestionType::Generic);
+        self.questions.insert(id, QuestionType::Base);
         Ok(object_path)
     }
 
     /// creates new specialized luks activation question without answer and password
```

Comment:
```suggestion
    /// creates new question to ask for a password with no default answer
```

Reply:
I think it is not right. There have to be always default strategy. I just want to mention here that created question has empty `answer` and `password` property.

Reply:
OK, then I did not understand the comment 😄